src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -461,8 +466,23 @@ public async Task RestartHostAsync(CancellationToken cancellationToken)
 
                     using (var activeOperation = ScriptHostStartupOperation.Create(cancellationToken, _logger))
                     {
-                        Task startTask = UnsynchronizedStartHostAsync(activeOperation);
-                        Task stopTask = Orphan(previousHost, cancellationToken);
+                        bool.TryParse(_config.GetSection(ConfigurationSectionNames.SequentialJobHostRestart).Value, out bool enforceSequentialOrder);
+                        Task startTask, stopTask;
+
+                        // If we are running in development mode with core tools, do not overlap the restarts.
+                        // Overlapping restarts are problematic when language worker processes are listening
+                        // to the same debug port
```

Comment:
+1 for  doing this only when running in core tools.  I believe we can add E2E tests in Core Tools for this change?

Reply:
I thought we were specifically tying to decouple this from core tools. What is the down side of the current implementation? I can see this being valuable in other scenarios as well, and it doesn't prevent us from automatically setting this from core tools and having tests there.

Reply:
Added a comment, the the port conflict is just one example of a resource that cannot be shared between processes. Others do apply. 

I would make the configuration name more intuitive (e.g. `DisableOverlappedRecycles`) or something along those lines.

Reply:
So far the only came up with in the context of local debugging. I cannot think of other scenarios where this flag would be useful outside of core tools. Do you have other examples where this might be useful?

Reply:
This could happen in any scenario where a resource contention might exist. From ports (like what we're seeing with the worker) to file locking issues (one example of that are some of the file locking issues that lead us to disable overlapped recycles in Antares). It's also good to keep in mind that, although we're only seeing this impact local debugging operations with _our_ scenarios, it would be entirely feasible that similar issues could impact customer workloads as well. 

This approach keeps the implementation generic, targeting a class of issues.

Reply:
Ok.  I see this flag is useful outside the context of language worker processes! Thanks.